lib/rabbitmq/cli/core/distribution.ex

```diff
@@ -69,36 +73,80 @@ defmodule RabbitMQ.CLI.Core.Distribution do
 
   defp start(node_name_type, attempts, _last_err) do
     candidate = generate_cli_node_name(node_name_type)
+
     case start_with_epmd(candidate, node_name_type) do
-      {:ok, _} -> :ok
-      {:error, {:already_started, pid}}      -> {:ok, pid};
-      {:error, {{:already_started, pid}, _}} -> {:ok, pid};
+      {:ok, _} ->
+        :ok
+      {:error, {:already_started, pid}} ->
+        {:ok, pid}
+      {:error, {{:already_started, pid}, _}} ->
+        {:ok, pid}
       {:error, reason} ->
         start(node_name_type, attempts - 1, reason)
     end
   end
 
   defp generate_cli_node_name(node_name_type) do
-    base                 = "rabbitmqcli" <> to_string(:rabbit_misc.random(100))
+    rmq_hostname = Helpers.get_rabbit_hostname()
```

Comment:
This will get a domain of the rabbitmq node addressed (`--node` argument or the env variable by default).
In case if you specify a node on a remote host with this argument, this function will return the remote host.
In theory you can do that, but I'm not sure of possible concequences.
Can it use the `RABBITMQ_NODENAME` env variable directly instead? It should contain the local nodename, because it's generated by scripts.

Reply:
I will take that into account. Thanks for pointing it out @hairyhum

Reply:
@hairyhum - actually, this does what we want already:

```
Helpers.get_rabbit_hostname()
```

Since no options are being passed to this function at this point, it will use either the value of `RABBITMQ_NODENAME` (if defined) or default to `rabbit@HOSTNAME` where `HOSTNAME` is the return value of the `hostname` function.

Reply:
Yes. Right. We probably could rename this function to be more clear.